source/common/http/http1/codec_impl.cc

```diff
@@ -962,8 +964,10 @@ void ClientConnectionImpl::onMessageComplete() {
   }
   if (pending_response_.has_value()) {
     // After calling decodeData() with end stream set to true, we should no longer be able to reset.
-    PendingResponse response = std::move(pending_response_.value());
-    pending_response_.reset();
+    PendingResponse& response = pending_response_.value();
+    // Encoder is used as part of decode* calls later in this function so pending_response_ can not
+    // be reset just yet. Preserve the state in pending_response_done_ instead.
+    pending_response_done_ = true;
```

Comment:
This doesn't thrill me due to extra logic complexity, but I think this is an OK solution for now. Can you do me a favor and put an additional `ASSERT(!pending_response_done_);` every time we load successfully check that pending_response has a value in the other functions above? I think there shouldn't be too many.

Reply:
Done